src/main/java/com/diffmin/util/SpoonUtil.java

```diff
@@ -73,19 +75,27 @@ public static CtModel buildModel(File file) throws FileNotFoundException {
         return launcher.buildModel();
     }
 
-    /**
-     * Pretty prints the model.
-     *
-     * @param model model to be pretty printed
-     * @return patched program
-     */
-    public static String displayModifiedModel(CtModel model) {
-        CtType<?> firstType = model.getAllTypes().stream().findFirst().get();
-        CtCompilationUnit cu = firstType.getFactory().CompilationUnit().getOrCreate(firstType);
+    public static String prettyPrintModel(CtModel model) {
+        List<CtCompilationUnit> compilationUnits =
+                List.copyOf(
+                        model.getUnnamedModule().getFactory().CompilationUnit().getMap().values());
+        List<CtCompilationUnit> modelCuCandidates =
+                compilationUnits.stream()
+                        .filter(cu -> !cu.getFile().getName().equals("package-info.java"))
+                        .collect(Collectors.toList());
+        if (modelCuCandidates.isEmpty()) {
+            throw new IllegalArgumentException(
+                    "Model does not have a compilation unit after excluding the package-info.java file");
+        }
```

Comment:
This case is covered by the next check, if `modelCuCandidates.isEmpty()` is true, then certainly `modelCuCandidates.size() != 1` is as well.